tensorflow/core/kernels/eigen_convolution_helpers.h

```diff
@@ -0,0 +1,81 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_CORE_KERNELS_EIGEN_CONVOLUTION_HELPERS_H_
+#define TENSORFLOW_CORE_KERNELS_EIGEN_CONVOLUTION_HELPERS_H_
+
+namespace Eigen {
+namespace internal {
+
+// TensorEvaluatorHasPartialPacket<TensorEvaluatorType, PacketType, IndexType>
```

Comment:
Thanks for this PR, very nice improvements! Could you please add few lines here, just with a high level overview how packing is done, and where this template is used, just to help the next reader of this code.

Reply:
> Thanks for this PR, very nice improvements! Could you please add few lines here, just with a high level overview how packing is done, and where this template is used, just to help the next reader of this code.

Added some comment on line 34.

Reply:
Added some comments on line 34.